packages/crypto-wallet-core/src/transactions/eth/index.ts

```diff
@@ -10,23 +14,48 @@ export class ETHTxProvider {
     gasLimit: number;
     network: string;
     chainId?: number;
+    contractAddress?: string;
   }) {
-    const { recipients, nonce, gasPrice, data, gasLimit, network } = params;
-    const { address, amount } = recipients[0];
+    const { recipients, nonce, gasPrice, gasLimit, network, contractAddress } = params;
+    let { data } = params;
+    let to;
+    let amount;
+    if (recipients.length > 1) {
+      if (!contractAddress) {
+        throw new Error('Multiple recipients requires use of multi-send contract, please specify contractAddress');
+      }
+      const addresses = recipients.map(recipient => recipient.address);
+      amount = 0;
+      const amounts = recipients.map(recipient => {
+        amount += recipient.amount;
```

Comment:
If amounts are passed in as a string, this will result in the strings being concatenated. To prevent giant number rounding, we should also use the BN lib from `web3.utils.BN`

Reply:
Here is what I have locally:
```
      const addresses = [];
      const amounts = [];
      const { toBN } = Web3.utils;
      amount = toBN(0);
      for (let recipient of recipients) {
        addresses.push(recipient.address);
        amounts.push(toBN(recipient.amount));
        amount = amount.add(toBN(recipient.amount));
      }
```